kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/service/impl/ProxyAuthServiceImpl.java

```diff
@@ -0,0 +1,648 @@
+			String redirectURI) {
+		// Compare states
+		if (!stateCookie.equals(state)) {
+			throw new AuthManagerException(AuthErrorCode.KEYCLOAK_STATE_EXCEPTION.getErrorCode(),
+					AuthErrorCode.KEYCLOAK_STATE_EXCEPTION.getErrorMessage());
+		}
+
+		HttpHeaders headers = new HttpHeaders();
+		headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+		MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
+		map.add(KeycloakConstants.GRANT_TYPE, loginFlowName);
+		map.add(KeycloakConstants.CLIENT_ID, clientID);
+		map.add(KeycloakConstants.CLIENT_SECRET, clientSecret);
+		map.add(KeycloakConstants.CODE, code);
+		map.add(KeycloakConstants.REDIRECT_URI, this.redirectURI + redirectURI);
+		Map<String, String> pathParam = new HashMap<>();
+		pathParam.put("realmId", realmID);
+		UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUriString(tokenEndpoint);
+		HttpEntity<MultiValueMap<String, String>> entity = new HttpEntity<>(map, headers);
+		ResponseEntity<String> responseEntity = null;
+		try {
+			responseEntity = restTemplate.exchange(uriBuilder.buildAndExpand(pathParam).toUriString(), HttpMethod.POST,
+					entity, String.class);
+
+		} catch (HttpClientErrorException | HttpServerErrorException e) {
+			KeycloakErrorResponseDto keycloakErrorResponseDto = parseKeyClockErrorResponse(e);
+			throw new LoginException(AuthErrorCode.KEYCLOAK_ACESSTOKEN_EXCEPTION.getErrorCode(),
+					AuthErrorCode.KEYCLOAK_ACESSTOKEN_EXCEPTION.getErrorMessage() + AuthConstant.WHITESPACE
+							+ keycloakErrorResponseDto.getError_description());
+		}
+		AccessTokenResponse accessTokenResponse = null;
+		try {
+			accessTokenResponse = objectmapper.readValue(responseEntity.getBody(), AccessTokenResponse.class);
+		} catch (IOException exception) {
+			throw new LoginException(AuthErrorCode.RESPONSE_PARSE_ERROR.getErrorCode(),
+					AuthErrorCode.RESPONSE_PARSE_ERROR.getErrorMessage() + AuthConstant.WHITESPACE
+							+ exception.getMessage());
+		}
+		AccessTokenResponseDTO accessTokenResponseDTO = new AccessTokenResponseDTO();
+		accessTokenResponseDTO.setAccessToken(accessTokenResponse.getAccess_token());
+		accessTokenResponseDTO.setExpiresIn(accessTokenResponse.getExpires_in());
+		return accessTokenResponseDTO;
+	}
+
+	private KeycloakErrorResponseDto parseKeyClockErrorResponse(HttpStatusCodeException exception) {
+		KeycloakErrorResponseDto keycloakErrorResponseDto = null;
+		try {
+			keycloakErrorResponseDto = objectmapper.readValue(exception.getResponseBodyAsString(),
+					KeycloakErrorResponseDto.class);
+		} catch (IOException e) {
+			throw new LoginException(AuthErrorCode.RESPONSE_PARSE_ERROR.getErrorCode(),
+					AuthErrorCode.RESPONSE_PARSE_ERROR.getErrorMessage() + AuthConstant.WHITESPACE + e.getMessage());
+		}
+		return keycloakErrorResponseDto;
+	}
+
+	@Override
+	public String getKeycloakURI(String redirectURI, String state) {
```

Comment:
Why are we assuming its keycloack? it can be any openid based system.

Reply:
Will change this to Unsupported operation